### ISSUE TITLE
lambda_sum_{largest,smallest} nd support

### ISSUE THREAD

**Issue by alexshtf** [cvxpy/cvxpy#3095]
An array of size $m \times n \times n$ can be thought of as a batch of $n \times n$ matrices. So I'd like to request that:
```python
x = cp.Variable(m, n, n, symmetric=True)
```
will be recognized as a batch of $m$ symmetric matrices, and
```python
cp.lambda_sum_largest(x, k)
```
will represent a vector of size $m$ where each entry is the sum of the $k$ largest eigenvalues of the corresponding matrix.

**Comment by Transurgeon** [comment]
thanks for the request! are there any applications of this batched lambda_sum_largest?
We'll think about how to add this and get back to you!

**Comment by alexshtf** [comment]
I am working on a DCCP (Disciplined Difference-of-Convex Programming) problem of fitting models to data, where my loss function over the samples involves `lambda_sum_largest`. So I don't want to iterate over my data samples, but instead I want a vectorized expression over the entire training set. Just like least-squares problems or logistic regression problems, where we have `cp.norm` and `cp.logistic` in vectorized form.

**Comment by PTNobel** [comment]
@alexshtf this and #3096 will be in CVXPY 1.9; which may not be fore a few more months. Feel free to install master in the meantime and smoke test the implementation!
